File: back/app/forms/sticky_note_form.py
from app.constants.sticky_note_constants import MAX_CONTENT_LENGTH
from app.lib.mixin.validatable import Validatable, model_name
from app.validations.blank_validation import BlankValidation
from app.validations.max_length_validation import MaxLengthValidation
from app.lib.mixin.validatable import validate


@model_name('sticky_note')
@validate('content', BlankValidation)
@validate('content', MaxLengthValidation, MAX_CONTENT_LENGTH)
class StickyNoteForm(Validatable):
    def __init__(self, **params):
        super().__init__()
        self.session_token = params.get('session_token')
        self.content = params.get('content')

File: front/src/models/forms/sticky-note-form.ts
import ApplicationForm from '@/models/forms/application-form';
import { blankValidation, maxLengthValidation } from '@/lib/validations';
import { MAX_CONTENT_LENGTH } from '@/constants/sticky-note';
import { NewStickyNoteParams } from '@/models/interfaces/api-params/sticky-notes';

export default class StickyNoteForm extends ApplicationForm {
  sessionToken!: string;
  content!: string;
  token!: string;

  constructor({
    sessionToken = '',
    content = '',
    token = ''
  }: { sessionToken?: string; content?: string; token?: string } = {}) {
    super();
    this.sessionToken = sessionToken;
    this.content = content;
    this.token = token;
  }

  initRules(): void {
    this.rules.set('sessionToken', this.sessionTokenRules());
    this.rules.set('content', this.contentRules());
  }

  sessionTokenRules(): Array<Function> {
    return this.generateRules(['StickyNote', 'sessionToken'], (messageKey: string) => {
      return [blankValidation(messageKey)];
    });
  }

  contentRules(): Array<Function> {
    return this.generateRules(['StickyNote', 'content'], (messageKey: string) => {
      return [blankValidation(messageKey), maxLengthValidation(messageKey, MAX_CONTENT_LENGTH)];
    });
  }

  createParams(): NewStickyNoteParams {
    return {
      session_token: this.sessionToken,
      content: this.content
    };
  }
}
